### PR TITLE
Add color highlighting for best and worst deals

### PR DESCRIPTION
**Pull request by EmergencyTemporalShift**
This lets you see the best deals going to or from a system.
I'm up to discussion for making this feature better.

### CODE REVIEW

**Comment by tehhowch** [comment]
Please review the Style Guide for code and data contributions:
http://endless-sky.github.io/styleguide/styleguide.xml

Of particular note are that you have used spaces for alignment, use in-line comments, have not appropriately spaced tokens and operators (i.e. `this==that` -> `this == that`, and appear to have simply copied code without actually determining if it is required or not.

Please also review the directly related issues: #2671, #1118

**Comment by EmergencyTemporalShift** [comment]
Thanks I somehow completely missed that. After I fix that, does this request stay open to discussion? I'm (perhaps clearly) new to this.

**Comment by EmergencyTemporalShift** [comment]
Alright, I replaced the spaces with tabs, changed the in-line comment, and removed the unnecessary code. I couldn't however, find the token spacing issue you were talking about, could you be more specific?

File: source/MapDetailPanel.cpp [hunk 1]
```diff
@@ -504,6 +520,12 @@ void MapDetailPanel::DrawInfo()
 			else
 			{
 				value -= localValue;
+				if(value==bestDeal) {
```
Comment (Bladewood) [inline source/MapDetailPanel.cpp]: > I couldn't however, find the token spacing issue you were talking about, could you be more specific?

> have not appropriately spaced tokens and operators (i.e. `this==that` -> `this == that`)

**Comment by EmergencyTemporalShift** [comment]
Ah, OK thanks. Should be fixed now.

**Comment by tehhowch** [comment]
Your use of braces is still not conformant with the game's style.
It isn't in the game's style guide, but is something actively being upgraded: arbitrary colors you are using should be pulled from the game's interfaces.txt definition, as otherwise they can not be modified by the end user.

On the subject of merit: given the issues I linked earlier, you're going to need to provide a good argument for adding this feature. I didn't link any issues that relate to it, but you can easily find issues or pulls that would have added colored text or colored outlines and were not implemented or asked to be changed.
For example, your use of pure red and pure green are liable to cause accessibility issues (red-green colorblindness) and have been generally considered to not fit well with the game's current [UI design](https://github.com/endless-sky/endless-sky/wiki/StyleGoals#ui-graphics).

As possible alternatives, consider drawing pointers, using the MapColor of the best and worst commodities instead of two arbitrary colors, etc.

At a bare minimum: this is a pull that adds or alters drawn elements: you need to provide screenshots of how this looks.

**Comment by endless-sky** [comment]
This is a suggestion that has been made several times in the past and rejected because colored text (especially in saturated primary colors) clashes with the design aesthetic of the UI.

**Comment by EmergencyTemporalShift** [comment]
Alright, closing now.

**Pull request closed without merging**